app/org/sagebionetworks/bridge/play/controllers/ScheduledActivityController.java

```diff
@@ -82,14 +84,23 @@ public Result updateScheduledActivities() throws Exception {
         UserSession session = getAuthenticatedAndConsentedSession();
 
         ScheduleContext.Builder builder = new ScheduleContext.Builder();
-        addEndsOnWithZone(builder, untilString, offset, daysAhead);
-        
+        // This zone is the zone of the request and the endsOn is scheduled relative to the user's
+        // time zone at the time of a request.
+        DateTimeZone zone = addEndsOnWithZone(builder, untilString, offset, daysAhead);
+
+        // This zone is the first zone used to schedule for the user, ever. If we don't have that:
+        // this request is the first contact from the user with a time zone, so save that.
+        DateTimeZone timeZone = session.getParticipant().getTimeZone();
+        if (timeZone == null) {
+            timeZone = persistTimeZone(session, zone);
+        }
```

Comment:
Rewrote these comments and renamed the context time zone to initialTimeZone (in contract to requestTimeZone).